src/test/scala/io/iohk/ethereum/ledger/BlockImportSpec.scala

```diff
@@ -196,7 +196,7 @@ class BlockImportSpec extends AnyFlatSpec with Matchers with ScalaFutures {
     // dying before updating the storage but after updating the cache, inconsistency is created
     blockchain.saveBestKnownBlocks(oldBlock4.number)
 
-    blockchainReader.getBestBlock() shouldBe Some(ancestorForValidation)
+    blockchainReader.getBestBlock() shouldBe Some(newBlock3)
```

Comment:
If it is simulating a bug that has currently disappeared, leaving the test is not a bad idea, because it will make sure that the bug does not come back in the following refactors

Reply:
Well, it's not simulating a bug in the sense that it check that the bug does not happens. It simulate the bug actually happening by putting the app in an inconsistent state and check that it returned the genesis (quite arbitrarily). But that is not the behavior anymore.
What worries me here is that it looks like we just changed the expected value to make the test pass. So I am not sure what the test is really testing now. But when we remove the `BlockchainMetadata` part it should become clearer, so let's leave it at that for now.